configs/recommended.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * @fileoverview the `recommended` config for `eslint.config.js`
+ * @author 唯然<weiran.zsd@outlook.com>
+ */
+
+"use strict"
+
+const mod = require("../lib/index.js")
+
+module.exports = {
+    plugins: { n: mod },
+    languageOptions: { globals: mod.configs.recommended.globals },
```

Comment:
Missing `sourceType`?

Reply:
I'm actually not sure if there should be `recommended` config because it would be wrong for `.cjs`/`.mjs` files. The current eslintrc config has [`overrides`](https://github.com/eslint-community/eslint-plugin-n/blob/bea9981a3993b239e51125c3b22e2534fe8d0496/lib/configs/recommended.js#L13-L16) for them, so it works well. We could do something similar here - export an array of configs with the `files` key - but that should generally be avoided in the new config system because it makes the config more difficult to manipulate with (e.g., to apply it to a specific directory only).

For commonjs projects, user's config should typically be:

```js
// user's eslint.config.js

const nodeRecommendedScript = require("eslint-plugin-n/configs/recommended-script");
const nodeRecommendedModule = require("eslint-plugin-n/configs/recommended-module");

module.exports = [
    {
        files: ["**/*.js", "**/*.cjs"],
        ...nodeRecommendedScript
    },
    {
        files: ["**/*.mjs"],
        ...nodeRecommendedModule
    }
]
```

For `"type": "module"` projects:

```js
// user's eslint.config.js

import nodeRecommendedScript from "eslint-plugin-n/configs/recommended-script.js";
import nodeRecommendedModule from "eslint-plugin-n/configs/recommended-module.js";

export default [
    {
        files: ["**/*.js", "**/*.mjs"],
        ...nodeRecommendedModule
    },
    {
        files: ["**/*.cjs"],
        ...nodeRecommendedScript
    }
];
```

What do you think?

Reply:
In the beginning, to minimize the difference with the eslintrc presets, I had wondered if a configuration could be provided:

```js
module.exports = packageJson.type === "module" ? moduleConfig : scriptConfig;
```

but it won't work well if it's using with "*.cjs" and "*.mjs". /_ \

Yes, this way sounds much clearer.  Will remove it and document the usage. Thanks for the thoughtful feedback! ❤️